### DISCUSSION TITLE
Why Found array with dim 3. Estimator expected <= 2?Use forecaster = PolynomialTrendForecaster(degree=1)

### DISCUSSION THREAD

**Discussion by ljf20122010** [alan-turing-institute/sktime#832] [Q&A]
I Use 
**forecaster = PolynomialTrendForecaster(degree=1)  ,**
and my y_trian is 1dim(**y_train_ndim=1**)
so why report Found array with dim 3. Estimator expected <= 2?
this is my code:

#-------------------Fitting an Auto Regressive Light-GBM------------
#Setting Quantile Regression Hyper-parameter.
params = {
    'objective':'quantile'
}
#A 10 percent and 90 percent prediction interval(0.1,0.9 respectively).
quantiles = [.1, .5, .9] #Hyper-parameter "alpha" in Light GBM
#Capture forecasts for 10th/median/90th quantile, respectively.
forecasts = []
#Iterate for each quantile.
for alpha in quantiles:
    

    forecaster = get_transformed_target_forecaster(alpha,params)
    
    #Initialize ForecastingHorizon class to specify the horizon of forecast
    fh = ForecastingHorizon(y_test.index, is_relative=False)
    
    #Fit on Training data.
    forecaster.fit(y_train)
    
    #Forecast the values.
    y_pred = forecaster.predict(fh)
    
    #List of forecasts made for each quantile.
    y_pred.index.name="date"
    y_pred.name=f"predicted_sales_q_{alpha}"
    forecasts.append(y_pred)

#Append the actual data for plotting.
store1_agg_monthly.index.name = "date"
store1_agg_monthly.name = "original"
forecasts.append(store1_agg_monthly)

**Comment by ljf20122010** [comment]
and the forecaster is :
    #----------------------------Create Pipeline--------------------
def get_transformed_target_forecaster(alpha,params):
```#Initialize Light GBM Regressor 

regressor = lgb.LGBMRegressor(alpha=alpha,**params)

#1.Separate the Seasonal Component.
#2.Fit a forecaster for the trend.
#3.Fit a Autoregressor to the resdiual(autoregressing on four historic values).

forecaster = TransformedTargetForecaster(
    [
        ("deseasonalise", Deseasonalizer(model="multiplicative", sp=12)),
        ("detrend", Detrender(forecaster=PolynomialTrendForecaster(degree=1))),
        (
            # Recursive strategy for Multi-Step Ahead Forecast.
            # Auto Regress on four previous values
            "forecast",
            RecursiveTimeSeriesRegressionForecaster(
                estimator=regressor, window_length=4
            ),
        ),
    ]
)
    return forecaster
```

**Reply by mloning** [reply]
Please post a [minimal, reproducible example](https://stackoverflow.com/help/minimal-reproducible-example)

**Reply by Hagen42** [reply]
Hey, I've got the same error.
At the line with _forecaster.fit(y)_ occurs following error: **Found array with dim 3. Estimator expected <= 2.**
I reproduced this error with the sktime dataset "airline". Following an example to reproduce:

```
import lightgbm as lgb

from sktime.datasets import load_airline
from sktime.transformations.series.detrend import Deseasonalizer, Detrender
from sktime.forecasting.trend import PolynomialTrendForecaster
from sktime.forecasting.compose import (
    TransformedTargetForecaster,
    RecursiveTimeSeriesRegressionForecaster,
)


def create_forecaster_w_desesonalizer(sp, degree=1):
    regressor = lgb.LGBMRegressor()
    forecaster = TransformedTargetForecaster(
        [
            ("deseasonalize", Deseasonalizer(model="additive", sp=sp)),
            ("detrend", Detrender(forecaster=PolynomialTrendForecaster(degree=degree))),
            (
                "forecast",
                RecursiveTimeSeriesRegressionForecaster(
                    regressor,
                    window_length=96,
                ),
            ),
        ]
    )
    **return** forecaster


y = load_airline()

forecaster = create_forecaster_w_desesonalizer(sp=12)

forecaster.fit(y)

print(forecaster.summary())
```

**Reply by fkiraly** [reply]
This looks like a bug. Moving to issue and bugfixing backlog.

**Reply by fkiraly** [reply]
issue here: https://github.com/alan-turing-institute/sktime/issues/2117
